### PR TITLE
Implemented `is_constructible` and related traits for MSVC 19

### PR DESCRIPTION
**Pull request by K-ballo**

### CODE REVIEW

File: test/type_traits.cpp [hunk 1]
```diff
@@ -355,74 +355,242 @@ TEST_CASE("is_unsigned", "[meta.unary.prop]") {
...
 }
 
 // template<class T> struct is_default_constructible;
 TEST_CASE("is_default_constructible", "[meta.unary.prop]") {
   CHECK(std::is_base_of<slb::true_type,
                         slb::is_default_constructible<int>>::value);
+
+  /* destructible */ {
+    struct ND {
+      ND(){};
```
Comment (mpark) [inline test/type_traits.cpp]: `ND() {}`

**Review by mpark — APPROVED** [review]

File: test/type_traits.cpp [hunk 2]
```diff
@@ -355,74 +355,242 @@ TEST_CASE("is_unsigned", "[meta.unary.prop]") {
...
 }
 
 // template<class T> struct is_default_constructible;
 TEST_CASE("is_default_constructible", "[meta.unary.prop]") {
   CHECK(std::is_base_of<slb::true_type,
                         slb::is_default_constructible<int>>::value);
+
+  /* destructible */ {
+    struct ND {
+      ND() {};
```
Comment (mpark) [inline test/type_traits.cpp]: The extraneous semi-colon is still here.

Comment (K-ballo) [reply test/type_traits.cpp]: Oh, so it wasn't (just) about spacing...

Comment (mpark) [reply test/type_traits.cpp]: 'twas both 😄

**Pull request merged**